src/clients/meta/MetaClient.cpp

```diff
@@ -948,6 +948,15 @@ void MetaClient::diff(const LocalCache& oldCache, const LocalCache& newCache) {
       for (const auto& oldPart : oldParts) {
         listener_->onPartRemoved(spaceId, oldPart.first);
       }
+
+      // wait spaceId all AdminTask stopped
+      if (taskListener_ != nullptr) {
+        taskListener_->cancelSpaceJobs(spaceId);
```

Comment:
Maybe only expose one interfaces like `waitSpaceJobs`, and we could do all the things in AdminTaskManager.